### PR TITLE
Flp 72 earnings report

### PR DESCRIPTION
**Pull request by JackWorkAccount**

### CODE REVIEW

File: src/SFA.DAS.ProviderFunding.Web/Services/CSVBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+            foreach (var obj in data.Learners)
+            {
+                Report.Add(new AcademicYearEarningsReport
+                {
+
+                    FamilyName = "FamilyName",
+                    GivenName = "GivenName",
+                    UinqueLearningNumber = obj.Uln,
+                    OnProgrammeEarnings_Jan = obj.OnProgrammeEarnings.Where(q => q.DeliveryPeriod == 1).Any() ? obj.OnProgrammeEarnings.Where(q => q.DeliveryPeriod == 1).First().Amount : 0,
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ProviderFunding.Web/Services/CSVBuilder.cs:22]: Would SingleOrDefault be better?

File: src/SFA.DAS.ProviderFunding.Web/Services/CSVBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace SFA.DAS.ProviderFunding.Web.Services
+{
+    public class CSVBuilder
+    {
+        public static List<AcademicYearEarningsReport> ExportToCSV(AcademicYearEarningsDto data)
+        {
+            var Report = new List<AcademicYearEarningsReport>();
+
+            foreach (var obj in data.Learners)
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ProviderFunding.Web/Services/CSVBuilder.cs:14]: Rename from obj

File: src/SFA.DAS.ProviderFunding.Web/Services/CSVBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using SFA.DAS.ProviderFunding.Web.Models;
+
+namespace SFA.DAS.ProviderFunding.Web.Services
+{
+    public class CSVBuilder
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ProviderFunding.Web/Services/CSVBuilder.cs:8]: It's not actually building a CSV in this class is it?

Comment (tom-gough) [reply src/SFA.DAS.ProviderFunding.Web/Services/CSVBuilder.cs:7]: no? and this is still the case now? should we update the name?

**Review by tom-gough — DISMISSED** [review]

**Review by dinathain — COMMENTED** [review]
I've added a few suggestions around casing/naming, plus a question on the Family/Given name generation (which I think I already know the answer to, but just checking first).

File: src/SFA.DAS.ProviderFunding.Web/Services/AcademicYearEarningsReportBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        public static List<AcademicYearEarningsReport> Build(AcademicYearEarningsDto data)
+        {
+            var Report = new List<AcademicYearEarningsReport>();
+
+            foreach (var learner in data.Learners)
+            {
+                Report.Add(new AcademicYearEarningsReport
+                {
+
+                    FamilyName = "FamilyName",
```
[the remark this answers is not in the document]
Comment (dinathain) [reply src/SFA.DAS.ProviderFunding.Web/Services/AcademicYearEarningsReportBuilder.cs:18]: This can be ignored - I can see there's another ticket for this.

**Review by dinathain — APPROVED** [review]
Approved - will just reference the comments in here in a tech debt ticket.

**Review by tom-gough — APPROVED** [review]
Approved because the decision was taken to resolve these comments in another ticket.

**Pull request merged**